pkg/cli/validate_test.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	v0 "github.com/abcxyz/jvs/apis/v0"
+	"github.com/abcxyz/jvs/pkg/config"
+	"github.com/abcxyz/pkg/testutil"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/spf13/cobra"
+)
+
+func TestRunValidateCmd(t *testing.T) {
+	// setup jwks server
+	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	keyID := "test_key_id"
+	ecdsaKey, err := jwk.FromRaw(privateKey.PublicKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := ecdsaKey.Set(jwk.KeyIDKey, keyID); err != nil {
+		t.Fatal(err)
+	}
+	jwks := make(map[string][]jwk.Key)
+	jwks["keys"] = []jwk.Key{ecdsaKey}
+
+	j, err := json.MarshalIndent(jwks, "", " ")
+	if err != nil {
+		t.Fatal("couldn't create jwks json")
+	}
+	path := "/.well-known/jwks"
+	mux := http.NewServeMux()
+	mux.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprintf(w, "%s", j)
+	})
+
+	svr := httptest.NewServer(mux)
+
+	t.Cleanup(func() {
+		svr.Close()
+	})
+
+	breakglassToken, err := v0.CreateBreakglassToken(testTokenBuilder(t), "testing")
+	if err != nil {
+		t.Fatalf("failed to build breakglass token: %s", err)
+	}
+
+	cfg = &config.CLIConfig{
+		JWKSEndpoint: svr.URL + path,
+	}
+
+	tests := []struct {
+		name            string
+		token           string
+		wantOut         string
+		wantErr         string
+		allowBreakglass bool
+	}{
+		{
+			name:    "signed",
+			token:   testSignToken(t, testTokenBuilder(t), privateKey, keyID),
+			wantOut: "Token is valid",
+		},
+		{
+			name:            "breakglass",
+			token:           breakglassToken,
+			wantOut:         "Token is valid",
+			allowBreakglass: true,
+		},
+		{
+			name:            "breakglass_not_allowed",
+			token:           breakglassToken,
+			wantErr:         "breakglass is forbidden, denying",
+			allowBreakglass: false,
+		},
+		{
+			name:    "invalid",
+			token:   "token",
+			wantErr: "failed to parse token headers",
+		},
+	}
+
+	for _, tc := range tests {
```

Comment:
IIUC, the tests can't paralleled because we are setting global flags? If so, let's add a comment explaining that.

Reply:
Could you clarify they "cannot" or "should not" run parallely with global flags? I am able to run and pass the tests with t.Parallel() therefore wondering.

Reply:
```
go test -count=1 -race -shuffle=on -parallel=20
```